### ISSUE TITLE
[proj 3] yield 가능 여부에 관한 질문

### ISSUE THREAD

**Issue by caelum02** [swsnu/osspr2022#100]
Project 3의 스펙에 따르면 task는 time slice를 소진한 뒤에 preemption이 가능하다고 합니다.
> When a task is scheduled, the task runs on a CPU until it uses up its time slice, and then is preempted.  
 
 만약 WRR Policy를 따르는 task A가  **time slice를 모두 소진하지 않은 상태**에서 **yield**한다면 스케쥴러는 yield 관련 시스템콜을 처리하는 과정에서 아래 중 하나의 행동을 하게 될 것 같습니다.

1. A를 다시 스케쥴함
2. A를 큐의 맨 뒤로 보내고, 큐의 맨 앞에 있는 태스크를 스케쥴함

둘 중 어느 것이 과제 스펙에 부합할까요?

**Comment by widiba03304** [comment]
2번 입니다. yield는 preemption policy와 상관없이 runqueue의 제일 뒤로 보내야 합니다.

**Comment by caelum02** [comment]
네 감사합니다. 그러면 yield 할 때는 타임슬라이스를 전부 소진하고 preempt된 것이 아니므로 타임슬라이스의 재계산은 수행되면 안되는 게 맞을까요?

**Comment by widiba03304** [comment]
@caelum02 네 맞습니다. yield시에는 재계산을 수행하지 않습니다.
